### ISSUE TITLE
[BUG] `routeFromHAR` doesn't work

### ISSUE THREAD

**Issue by DetachHead** [microsoft/playwright#15165]
**Context:**
## System:
 - OS: Windows 10 10.0.19044
 - Memory: 1.41 GB / 15.78 GB
## Binaries:
 - Node: 18.0.0 - C:\Program Files\nodejs\node.EXE
 - npm: 8.7.0 - C:\Program Files\nodejs\npm.CMD
## Languages:
 - Bash: 5.0.17 - C:\WINDOWS\system32\bash.EXE
## npmPackages:
 - playwright: ^1.23.0 => 1.23.0

**Code Snippet**
```ts
// playwright.config.ts
import { PlaywrightTestConfig } from '@playwright/test'
const config: PlaywrightTestConfig = {
    projects: [
        {
            name: 'Chromium',
            use: {
                contextOptions: {
                    recordHar: { mode: 'full', path: 'asdf.har' },
                },
                browserName: 'chromium',
            },
        }
    ],
}
export default config
```
1. first run this to generate the har file:
```javascript
test('test', async ({ page }) => {
    await page.goto('https://google.com/')
    await page.fill('input[aria-label="Search"]', 'asdf')
});
```
2. disable your internet connection to verify that it's using the responses from the har file
3. run the same test again with `await page.routeFromHAR('asdf.har')` at the beginning
```javascript
test('test', async ({ page }) => {
    await page.routeFromHAR('asdf.har')
    await page.goto('https://google.com/')
    await page.fill('input[aria-label="Search"]', 'asdf')
});
```

**Describe the bug**

the site doesn't load and instead hangs on a blank screen:

![image](https://user-images.githubusercontent.com/57028336/176072647-d74e1960-f188-498e-b998-11b337434ff3.png)

**Comment by rwoll** [comment]
I can half reproduce, and have a suggested fix. 😄 It looks like `routeFromHar` works once after the initial recording, and then the next run it hangs. 

However, if I remove the following:

```
                contextOptions: {
                    recordHar: { mode: 'full', path: 'asdf.har' },
                },
```

after the initial recording, **the `routeFromHar` variant works repeatedly.** 

The reason the above options break the example in https://github.com/microsoft/playwright/issues/15165#issue-1286657827 is they overwrite the HAR with an empty/corrupt HAR.

To get the initial HAR baseline, I suggest using the CLI, or in your config, use a different path from where you will be serving the HARs to `page.routeFromHar`. Before trying again with this info, completely delete your existing har.

`routeFromHar` is a brand new API, so there's still some iteration on the workflows to be done. **I'll bring this issue up with the team tomorrow. Thanks!**

As an aside: depending on the complexity of the site and how it's implemented, Playwright may not work with routeFromHar and more advanced routing handlers may be needed. This doesn't appear to be the case for why this particular example is broken, but do note, Playwright now has more route handling options including chaining handlers which it could not do before.

**Comment by unlikelyzero** [comment]
@rwoll @DetachHead in this case, should we do something like this:

```js
//package.json
 "test:pw:generatehar": "npx playwright test --config=playwright.config.js --project=generate-har",
 "test:pw":"npx playwright test --config=playwright.config.js --project=local-chrome",
```

```js
//playwright.config.js
      {
        name: "local-chrome",
          use: {
              browserName: 'chromium',
              channel: 'chrome',
          },
      },
      {
        name: "generate-har",
          use: {
              browserName: 'chromium',
              channel: 'chrome',
              contextOptions: {
                recordHar: { mode: 'full', path: 'asdf.har' },
            },
          },
      },
 ```

**Comment by rwoll** [comment]
If possible, hold off for now! I'll comment back with the recommended workflow in the coming days. We're experimenting in making the workflow friendlier out of the box to avoid reaching for these configs unless it's an advanced use case.

tl;dr: We're experimenting[^1] treating HAR's like we do screenshots/snapshots, so a similar update mechanism could be used. (i.e. you wouldn't have to toggle your code except for maybe on flag that would also be exposed via a command line switch).

[^1]: i.e. subject to change before we land and release the updated workflow

**Comment by pavelfeldman** [comment]
You should not use `recordHar: { mode: 'full', path: 'asdf.har' }` in your config - it would try to record har for every time, concurrently if there are several workers, into one file.

**Comment by rwoll** [comment]
Closing per above as part of triage. ✨ **v1.23.1** ✨  (released today) now has an **`update`** option which you can toggle when calling `page.routeFromHar` to trigger recording the HAR: https://playwright.dev/docs/next/api/class-page#page-route-from-har-option-update.

So,

1. Drop the `recordHar` option in your config; you shouldn't need it, and it causes [problems](https://github.com/microsoft/playwright/issues/15165#issuecomment-1171642522) even if you're not doing har routing
2.  Change your test to look like:
     ```ts
    test('test', async ({ page }) => {
        await page.routeFromHAR('test-1.har', { update: !!process.env.REBASELINE_HAR })
        await page.goto('https://google.com/')
        await page.fill('input[aria-label="Search"]', 'asdf')
    });
     ```

And then when you want to record, run like:

```
REBASELINE_HAR=1 npx playwright test 
```
